xml/System.Runtime.CompilerServices/MethodImplOptions.xml

```diff
@@ -135,7 +135,9 @@
       </ReturnValue>
       <MemberValue>512</MemberValue>
       <Docs>
-        <summary>The method contains a hot path and should be optimized.</summary>
+        <summary>The method should skip tiered-compilation and immediately reach tier0. 
+                 This can be beneficial for certain hot-path code which does not rely on any static state or any optimisations applied at re-jit time. 
+                 This attribute will not result in faster or better optimised code and is designed for use in very specific benchmark-lead scenarios.</summary>
       </Docs>
```

Comment:
This describes what this attribute does in the current coreclr runtime. I expect that the details will change a lot over time. We typically refrain from documenting implementation details like this since it makes the documentation constantly outdated.

> The method contains a hot path and should be optimized

All methods are optimized. I would change the description to "The method contains a very critical hot path and should be optimized with highest optimization level possible.".

But I think we should avoid describing the current implementation details of coreclr runtime.

Reply:
Agree with Jan. 

I also wouldn't connect it to benchmarking.